frontend/src/Frontend/UI/Dialogs/AddAccount.hs

```diff
@@ -7,34 +8,53 @@ module Frontend.UI.Dialogs.AddAccount
   ) where
 
 import           Control.Lens
+import           Data.Text (Text)
 ------------------------------------------------------------------------------
 import           Reflex
 import           Reflex.Dom
 ------------------------------------------------------------------------------
 import           Frontend.Wallet
-import           Frontend.Network
 import           Frontend.UI.DeploymentSettings (transactionDisplayNetwork, userChainIdSelect)
 
+import           Frontend.Ide (ide_wallet)
 import           Frontend.UI.Modal
+import           Frontend.UI.Modal.Impl (ModalIde)
 import           Frontend.UI.Widgets
 import           Frontend.Foundation
 
-type HasAddAccountModelCfg mConf key t =
+type HasAddAccountModelCfg model mConf key t =
   ( Monoid mConf, Flattenable mConf t
   , HasWalletCfg mConf key t
   )
 
 uiCreateWalletOnlyAccount
   :: forall t m model key mConf
      . ( MonadWidget t m
-       , HasAddAccountModelCfg mConf key t
-       , HasNetwork model t
+       , HasAddAccountModelCfg model mConf key t
        )
-  => model
+  => ModalIde m key t
   -> Event t ()
   -> m (mConf, Event t ())
-uiCreateWalletOnlyAccount model _onClose = do
-  onClose <- modalHeader $ text "Add Account"
+uiCreateWalletOnlyAccount model onClose = mdo
+  onClose <- modalHeader $ dynText title
```

Comment:
Shadows argument

Reply:
Something very weird happened here... The code you commented on did indeed shadow that variable. However my local version did not as the input to the function was named `_onClose` not `onClose`.

Updating this code to rename the variable to `onCloseExternal` now all of a sudden shows the input to the function as having been  `_onClose` and not `onClose`. :/